addons/mail/static/src/core/common/message.js

```diff
@@ -191,6 +189,15 @@ export class Message extends Component {
                 this.message.body,
             ]
         );
+        this.debouncedPrepareMessageBody = useDebounced(this.prepareMessageBody, 50);
```

Comment:
Why is it a problem to let it render twice?
If this is related to a problem you had to click on Save twice, this is fixed by this PR: https://github.com/odoo/odoo/pull/178066

I would not implement with such debounce, this is arbitrary and increases technical debts because this is prone to race-condition and this is a workaround with hacky code. Is there a genuine issue with having 2 renders rather than 1? If not, I wouldn't bother to try and improve with such a debounce.

Reply:
The scenario i mentioned would cause the `useEffect` to be called twice. First, for `isEditing` and then `message.body`. So, in this scenario the first render is useless, as the later one is the accurate one.

Reply:
@rapr-odoo This makes sense to have 2 renderings, and 50ms won't prevent it.
The state.isEditing changes quite immediately, while the message.body is payload from server. It works locally with 50ms on your machine, but the RPC and/or bus notification can take seconds.

So there's really no need to batch them with debounce, let it render twice if need be.